reference-sheets/GitPractices.md

```diff
@@ -0,0 +1,204 @@
+
+Removing your changes from the stash and placing them back in your working changes is called **popping** the changes. Popping your stashed working changes throws them back into your code, like popping a bubble full of working changes, and once it's popped the working changes fall back onto the code.
+
+Two people on a branch is not the only situation when you'd use stash. Sometimes, you want to write something a different way, but want to keep your changes just in case you change your mind. In this scenario, you can just stash your working changes, then rewrite it all, and if you don't like the rewrite, all you have to do is pop! Stashed changes restored. 
+
+### Merging
+
+When you're finally done with your changes on a branch, and you are sure that it is absolutely perfect, you create something called a **Pull Request** on Github, to merge it to the main branch and be added to the main codebase. Someone with merge permissions (an admin of the repository) will look over your code, indirectly humble your coding skills, and then tell you what to change before your code can be added to main. After the person reviewing your code thinks that it is perfect, they will approve it and you can then **merge your code into main**, bringing your changes officially into the codebase!
+
+Essentially, merging is how you transfer code from **branch to branch**. This, of course, will include transferring code **from** main **to** your branch, to stay updated with recent changes to main. If other people make a pull request and it gets approved, then the main branch will have some new code that is **not** currently in your working branch. This means that you are **not** up to date with main. Therefore, you need to **merge main into your branch** every now and again, and especially before you make a pull request, to make sure that everything is up to date. When you merge main into your branch, sometimes there will be conflicts, especially if the file being edited by both you and the other contributor are the same. These are called **Merge Conflicts**, and can happen both with merging **and** with pulling changes from a branch. You just have to sort these out manually, deciding which changes to keep and which changes to toss. It's a little annoying to do, but if you do it often enough and efficiently enough then it doesn't become that big of a problem.
+
+There is another way to merge main into your code called **Rebasing**, but it's complicated to explain in writing and you'll never use it with a team as small as a high school robotics team. The idea behind it is that you rewrite your commit history during a rebase, and that **instead** of branching off of the main branch that was four commits ago, you are **now** technically branched off of the newest main commit. It's useful for organizing commits to turn the visualization from a web to a tree, but when you're doing such small changes to a tiny codebase like used in robotics, it doesn't even matter. 
+
```

Comment:
Considering this seems to be the 'git' ref sheet for now, you might want to mention the .gitignore, what it does, and why it's useful. Could also just link to the GitHub/git docs on it and call it a day; they're pretty in-depth.

Reply:
just linked to git docs